pkg/cloud/awscloud/awscloud.go

```diff
@@ -26,6 +26,24 @@ type AWS struct {
 	s3       *s3.S3
 }
 
+type S3Permission string
+
+const (
+	S3PermissionRead        S3Permission = s3.PermissionRead
+	S3PermissionWrite       S3Permission = s3.PermissionWrite
+	S3PermissionFullControl S3Permission = s3.PermissionFullControl
+	S3PermissionReadAcp     S3Permission = s3.PermissionReadAcp
+	S3PermissionWriteAcp    S3Permission = s3.PermissionWriteAcp
+)
+
+var PermissionsMatrix = map[S3Permission][]S3Permission{
+	S3PermissionRead:        {S3PermissionRead, S3PermissionWrite, S3PermissionFullControl},
+	S3PermissionWrite:       {S3PermissionWrite, S3PermissionFullControl},
+	S3PermissionFullControl: {S3PermissionFullControl},
+	S3PermissionReadAcp:     {S3PermissionReadAcp, S3PermissionWriteAcp},
+	S3PermissionWriteAcp:    {S3PermissionWriteAcp},
+}
```

Comment:
Took me a minute to realise what this mapping was for.  A comment here or a docstring on the public method would help.

Reply:
I thought so too. There have been so few comments in the whole file but that's probably subject to change :-)